### DISCUSSION TITLE
Access is denied for anonymous user

### DISCUSSION THREAD

**Discussion by Griffith88** [apache/superset#25299] [Q&A / Help] [answered]
i have problem with public (no login) access to dashboards
I created role Guest and added in superset_config.py

![image](https://github.com/apache/superset/assets/42404689/a97e1fd0-e097-4298-8108-0793197e1d85)

```py
PUBLIC_ROLE_LIKE = "Guest"

DEFAULT_FEATURE_FLAGS = {
    "DASHBOARD_RBAC": True,
    "ENABLE_TEMPLATE_PROCESSING": True,
    "DASHBOARD_NATIVE_FILTERS": True,
    "DASHBOARD_CROSS_FILTERS": True
}
```
when i publish and share via link any dashboard or chart i get Access is Denied and its redirect me on Log In Page.

![image](https://github.com/apache/superset/assets/42404689/4fd079e9-adc8-4714-bd86-2d85d8150e01)

Any suggestions how to evade it ?

**Comment by Griffith88** [comment, marked as answer]
After several days i have finally found a solution to share dashboards through menu or via permalink.

Superset version: 2.1.0

superset_config.py 
```py
AUTH_ROLE_PUBLIC = 'Public'
PUBLIC_ROLE_LIKE = "Guest"  # i created guest role for anonymous view dashboards

DEFAULT_FEATURE_FLAGS = {
    "DASHBOARD_RBAC": True,  # it's obligatory
    "ENABLE_TEMPLATE_PROCESSING": True,
    "DASHBOARD_NATIVE_FILTERS": True,
    "DASHBOARD_CROSS_FILTERS": True
}
```

it's base configuration for all options.

permissions for guest role just for viewing dashboards:
![image](https://github.com/apache/superset/assets/42404689/4b8f9e21-87c6-4c22-ab67-df7c1b6cfb7c)

_menu access on Dashboards_ is optional

_can read on DashboardPermalinkRestApi_ ,  _can dashboard permalink on Superset_ , _can read on ExplorePermalinkRestApi_ is obligatory if u want to share dashboard via permalink:
![image](https://github.com/apache/superset/assets/42404689/7b4aa017-be33-4e4f-aaf1-4c2f88037ed9)

**DASHBOARD PROPERIES**:
![image](https://github.com/apache/superset/assets/42404689/9f90f279-0410-43e6-8522-415d4d0c186e)

it's obligatory to set URL SLUG if u want to access dashboard through menu

it's obligatory to set ROLES, definition is wrong.  If u leave this row empty it will not work.

**Comment by DevIA05** [comment]
Hello, thank you for sharing your work

I use superset with docker. I did what you did but I still get the **access denied** page when I enter the url generated by *copy permalink to clipboard*

```
git clone https://github.com/apache/superset.git
cd superset
git checkout 3.0.0
set TAG=3.0.0 
// I add the elements indicated in the file docker/pythonpath_dev/superset_config.py
docker compose -f docker-compose-non-dev.yml up -d
```

*docker/pythonpath_dev/superset_config.py* file
[superset_config.zip](https://github.com/apache/superset/files/13111033/superset_config.zip)


In superset, 
- I create *guest* role 
![image](https://github.com/apache/superset/assets/111973830/7c646d53-9b68-4273-a4f5-87eb4aaa9e0b)
- I edit dashboard properties
![image](https://github.com/apache/superset/assets/111973830/3ed451a6-c74f-4bd3-ac81-1ba394100fe7)

**Comment by Griffith88** [comment]
@DevIA05  Hi! As i see you didn't set all necessary permissions which i exposed on screenshot. It's obligatory .
After you have created new role Guest via web interface i recommend you to step into docker container `docker exec -it <your_container> /bin/bash` with superset_app and execute command `superset init`, so Public role will synchronize with Guest role. If all will be correct -> Public role will have the same permissions as Guest role

**Reply by DevIA05** [reply]
Hello, 
I forgot to mention it, but I don't have this permissions,  when I write I get the message: *No results found*
-  can recent activity on Superset 
-  can favstar on superset
- can slice json on Superset
- can result on Superset

**Reply by Griffith88** [reply]
Hello! And i forgot to mention that I use  2.1.0 version of superset. So far I can advice to you is to play with different permissions. 
I copied all permissions from Gamma role  to Guest role and then started shrinking them =)
